packages/rum-core/test/common/event-target-patch.spec.js

```diff
@@ -0,0 +1,142 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import patchEventHandler from './patch'
+import { EVENT_TARGET } from '../../src/common/constants'
+
+describe('EventTargetPatch', function() {
+  let events = []
+  let cancelFn
+
+  beforeAll(function() {
+    cancelFn = patchEventHandler.observe(EVENT_TARGET, function(event, task) {
+      events.push({
+        event,
+        task
+      })
+    })
+  })
+
+  afterAll(function() {
+    cancelFn()
+  })
+
+  beforeEach(function() {
+    events = []
+  })
+
+  it('should work different argument types', () => {
+    let element = document.createElement('div')
+    element.addEventListener('click', undefined)
+    var spy = jasmine.createSpy('spy')
+    element.addEventListener('click', spy)
+    element.click()
+    expect(spy).toHaveBeenCalled()
+  })
+
+  it('should not affect other event types', () => {
+    const eventType = 'apm-test-event'
+    var event = new Event(eventType)
+    let count = 0
+    let element = document.createElement('div')
+    const listener = e => {
+      expect(e.type).toBe(eventType)
+      count++
+    }
+    element.addEventListener(eventType, listener)
+    element.dispatchEvent(event)
+    expect(count).toBe(1)
+    expect(events.length).toBe(0)
+    element.removeEventListener(eventType, listener)
+    element.dispatchEvent(event)
+    expect(count).toBe(1)
+  })
+
+  if (window.EventTarget) {
+    it('should patch addEventListener', () => {
+      let count = 0
+      let element = document.createElement('div')
+      const listener = e => {
+        expect(e.type).toBe('click')
+        count++
+      }
+
+      element.addEventListener('click', listener)
+      element.addEventListener('click', listener, { capture: true })
+      element.click()
+      expect(count).toBe(2)
+      element.removeEventListener('click', listener)
+      element.click()
+      expect(count).toBe(3)
+      element.removeEventListener('click', listener, true)
+      element.click()
+      expect(events.map(e => e.event)).toEqual([
+        'schedule',
+        'invoke',
+        'schedule',
+        'invoke',
+        'schedule',
+        'invoke'
+      ])
+      expect(count).toBe(3)
+    })
+
+    it('should consider event listener uniqueness condition', () => {
+      let count = 0
+      let element = document.createElement('div')
+      const listener = e => {
+        expect(e.type).toBe('click')
+        count++
+      }
+
+      element.addEventListener('click', listener)
+      element.addEventListener('click', listener)
+      element.addEventListener('click', listener, { capture: false })
```

Comment:
needs test for passive event listeners.

Reply:
@vigneshshanmugam , I've looked into `passive` and `once` options, it still doesn't seem to be relevant to us to test and given that these are not supported on most of the browsers we test on (even chrome have added it in 51), I don't see a lot of benefit in adding these at this point.

Reply:
We can do test like running them only on browsers supporting it. I do not want to block the PR for this, but it would be great to have the tests anyway.